src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -711,18 +711,34 @@ void grpc_chttp2_encode_header(grpc_chttp2_hpack_compressor* c,
   }
   for (size_t i = 0; i < extra_headers_size; ++i) {
     grpc_mdelem md = *extra_headers[i];
-    uintptr_t static_index = grpc_chttp2_get_static_hpack_table_index(md);
-    if (static_index) {
-      emit_indexed(c, static_cast<uint32_t>(static_index), &st);
+    const bool is_static =
+        GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC;
+    if (is_static) {
+      uintptr_t static_index =
+          reinterpret_cast<grpc_core::StaticMetadata*>(GRPC_MDELEM_DATA(md))
+              ->StaticIndex();
+      if (static_index < GRPC_CHTTP2_LAST_STATIC_ENTRY) {
+        emit_indexed(c, static_cast<uint32_t>(static_index + 1), &st);
+      } else {
+        hpack_enc(c, md, &st);
+      }
```

Comment:
since `else {...}` block here is a dup of the `else` below, I would write the if statement as:
```
if (is_static && (static_index = ....->StaticIndex() < LAST_ENTRY)) {
  ...
} else {
 hpack_enc(...);
}